tests/e2e/operator/util/gomega/matchers.go

```diff
@@ -0,0 +1,87 @@
+package matchers
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/onsi/gomega/types"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// HaveConditionMatcher checks for a specific condition and status in a Kubernetes object.
+type HaveConditionMatcher struct {
+	conditionType      string
+	conditionStatus    metav1.ConditionStatus
+	lastSeenConditions []string // To store the last seen conditions for error reporting
+}
+
+// HaveCondition creates a new HaveConditionMatcher.
+func HaveCondition(conditionType string, conditionStatus metav1.ConditionStatus) types.GomegaMatcher {
+	return &HaveConditionMatcher{
+		conditionType:   conditionType,
+		conditionStatus: conditionStatus,
+	}
+}
```

Comment:
Since we want to pass in different conditions instead of `string`, I think we'll have to use generics. I believe this should work:
```suggestion
func HaveCondition[T ~string](conditionType T, conditionStatus metav1.ConditionStatus) types.GomegaMatcher {
	return &HaveConditionMatcher{
		conditionType:   string(conditionType),
		conditionStatus: conditionStatus,
	}
}
```